contracts/op/OPReverseVerifier.sol

```diff
@@ -1,74 +1,83 @@
-		return uint256(RLPReaderExt.bytes32FromRLP(v[8]));
-	}
+    constructor(
+        string[] memory urls,
+        uint256 window,
+        IVerifierHooks hooks,
+        IL1Block l1Block
+    ) AbstractVerifier(urls, window, hooks) {
+        _l1Block = l1Block;
+    }
 
-	function getStorageValues(bytes memory context, GatewayRequest memory req, bytes memory proof) external view returns (bytes[] memory, uint8 exitCode) {
-		uint256 blockNumber1 = abi.decode(context, (uint256));
-		GatewayProof memory p = abi.decode(proof, (GatewayProof));
-		RLPReader.RLPItem[] memory v = RLPReader.readList(p.rlpEncodedL2Block);
-		bytes32 blockHash = blockhash(_extractBlockNumber(v));
+    function getLatestContext() external view returns (bytes memory) {
+        return abi.encode(_l1Block.number());
+    }
 
-		// console2.logBytes32(blockHash);
-		// console2.logBytes32(RLPReaderExt.strictBytes32FromRLP(v[19]));
-		// (, bytes memory u) = BEACON_ROOTS_ADDRESS.staticcall(abi.encode(RLPReaderExt.bytes32FromRLP(v[11])));
-		// console2.logBytes(u);
+    struct GatewayProof {
+        bytes rlpEncodedL1Block;
+        bytes rlpEncodedL2Block;
+        bytes accountProof;
+        bytes storageProof;
+        bytes[] proofs;
+        bytes order;
+    }
 
-		require(blockHash == keccak256(p.rlpEncodedL2Block), "ReverseOP: hash2");
-		bytes32 stateRoot = RLPReaderExt.strictBytes32FromRLP(v[3]);
-		bytes32 storageRoot = EthTrieHooks.proveAccountState(stateRoot, address(_oracle()), p.accountProof);
-		blockHash = EthTrieHooks.proveStorageValue(storageRoot, address(0), SLOT_HASH, p.storageProof);
-		require(blockHash == keccak256(p.rlpEncodedL1Block), "ReverseOP: hash1");
-		v = RLPReader.readList(p.rlpEncodedL1Block);
-		_checkWindow(blockNumber1, _extractBlockNumber(v));
-		stateRoot = RLPReaderExt.strictBytes32FromRLP(v[3]);
-		return GatewayProver.evalRequest(req, ProofSequence(0, 
-			stateRoot,
-			p.proofs, p.order,
-			EthTrieHooks.proveAccountState,
-			EthTrieHooks.proveStorageValue
-		));
-	}
+    function getStorageValues(
+        bytes memory context,
+        GatewayRequest memory req,
+        bytes memory proof
+    ) external view returns (bytes[] memory, uint8 exitCode) {
+        uint256 blockNumber1 = abi.decode(context, (uint256));
+        GatewayProof memory p = abi.decode(proof, (GatewayProof));
+        RLPReader.RLPItem[] memory v = RLPReader.readList(p.rlpEncodedL2Block);
+        bytes32 blockHash = blockhash(_extractBlockNumber(v));
+        require(
+            blockHash == keccak256(p.rlpEncodedL2Block),
+            'ReverseOP: hash2'
+        );
+        bytes32 stateRoot = RLPReaderExt.strictBytes32FromRLP(v[3]);
+        bytes32 storageRoot = _hooks.verifyAccountState(
+            stateRoot,
+            address(_l1Block),
+            p.accountProof
+        );
+        blockHash = _hooks.verifyStorageValue(
+            storageRoot,
+            address(0),
+            SLOT_HASH,
```

Comment:
What is this?

Reply:
* shorthand but it looks funny when formatted
* `EthVerifierHooks` ignores `target`
* I'll change it to `address(_l1Block)` to be explicit
* that's the proof for slot 2 (blockhash) at a prior block